JSQMessagesViewController/Controllers/JSQMessagesViewController.m

```diff
@@ -269,6 +269,17 @@ - (void)viewDidLoad
 
     [self jsq_configureMessagesViewController];
     [self jsq_registerForNotifications:YES];
+    
+    // Per https://twitter.com/steipete/status/219848151765483520?lang=en
+    // "Is there some trick to bring UINavigationController to not mess with a UIScrollView  (_adjustContentOffsetIfNecessary)?"
+    // "Add a dummy view as first subview. It won’t look further."
+    //
+    // This hack prevents UINavigationController from calling _adjustContentOffsetIfNecessary
+    // on the collection view after viewWillAppear: but before viewDidAppear:, which
+    // causes the scroll state to reset to zero.    
+    UIView *dummyView = [UIView new];
+    dummyView.hidden = YES;
+    [self.view insertSubview:dummyView atIndex:0];
```

Comment:
I did an awkward and imperfect workaround for this in the Spring. This seems like a much better solution.

Reply:
To be sure I understand - this change intends to keep us from losing our specified scroll position?

Reply:
That's right. For some unknown reason, UINavigationController tries to reset your scroll state to zero/origin when presenting a view.